### DISCUSSION TITLE
v9.0.1 Core not finding installed emane bindings

### DISCUSSION THREAD

**Discussion by derkin123** [coreemu/core#721] [Q&A] [answered]
I have installed 9.0.1 core and 1.2.5 emane together following the wiki instructions, and I have installed the emane python bindings aswell using `inv install-emane -e v.1.2.5`. When trying to run a test emane network in the core-gui (just one node linked to an emane net) I get an error in the daemon that the bindings are not installed. The call of
`from emane.events import EventService, PathlossEvent, CommEffectEvent, LocationEvent`
in emanemanager.py throws an error of `no module named 'emane'.`

If I run poetry shell in .../core/daemon/ and call import emane I get no errors, if I try one step further and import emane.events I get this error:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/.cache/pypoetry/virtualenvs/core-7tW4151H-py3.9/lib/python3.9/site-packages/emane/events/__init__.py", line 36, in <module>
    from .eventservice import EventService
  File "/home/user/.cache/pypoetry/virtualenvs/core-7tW4151H-py3.9/lib/python3.9/site-packages/emane/events/eventservice.py", line 34, in <module>
    from . import event_pb2
  File "/home/user/.cache/pypoetry/virtualenvs/core-7tW4151H-py3.9/lib/python3.9/site-packages/emane/events/event_pb2.py", line 5, in <module>
    from google.protobuf import descriptor as _descriptor
ModuleNotFoundError: No module named 'google'

I don't quite know why core ins't able to spot the emane python bindings. Running poetry show doesn't seem to list emane as a package (although I don't know if it would show up here if everything was working), although running poetry shell then pip list shows emane version of 1.2.5. Any thoughts would be greatly appreciated!

**Comment by bharnden** [comment]
What type of install did you do exactly for CORE?

For `inv install-emane`, I should have expanded on the documentation. You would need to provide the same and proper PYTHON binary to use for that step, so it would be built and placed into the correct python interpreter.

You can view the CentOS Dockerfile as a good example of what exactly needs to be done: https://github.com/coreemu/core/blob/master/dockerfiles/Dockerfile.centos

Depending on the OS being used, you may also need a more recent `protoc` for the EMANE portion, which is also noted in the Dockerfile example above.

**Reply by derkin123** [reply]
I installed CORE through the scripts provided (cloning the repo, running ./setup.sh, and then running inv install). Also I am running this on Ubuntu 18. I don't believe that the python binary being used is wrong as I relinked python3 to point to python3.9 so /usr/bin/python3 points to /usr/bin/python3.9. My current protobuf version installed is 3.12.3 and my grpc is 1.49.0

**Comment by derkin123** [comment]
So I think it could be that I need to upgrade protobuf to at least 3.19. I followed the same steps in the docker example to install the bindings as opposed to the steps the install_emane function runs through (the main difference is the last step of /opt/core/venv/bin/python -m pip install . in the docker and poetry run pip install <emane python dir> in the install_emane), and I made sure to make my PYTHON=/opt/core/venv/bin/python. After doing this running core-daemon yields an error about the incorrect protoc version being used to generated files:

![image](https://user-images.githubusercontent.com/100057696/204892489-6c93bf0c-c0e6-45a7-9a33-10d7695c094d.png)

**Reply by bharnden** [reply]
Yes, you can test with `protoc --version` or whatever their version command is to verify, But, then follow the steps to download a prebuilt and prefix your PATH with the new one, so it gets picked up first.

**Comment by derkin123** [comment, marked as answer]
So I was able to get it to run after updating to protobuf 3.19.0. For anyone who lands here in the future to install the emane bindings properly on Ubuntu 18 I did the following:

a.	cd <core_repo>/..
b.	git clone https://github.com/adjacentlink/emane.git -b <vX.X.X>
c.	cd emane
d.	./autogen.sh
e.	PYTHON=/opt/core/venv/bin/python ./configure --prefix=/usr
f.	cd src/python
g.	make
h.	sudo /opt/core/venv/bin/python -m pip install .
